client/src/components/LoginForm/LoginContent.jsx

```diff
@@ -1,19 +1,120 @@
 import React, { Component } from 'react';
 import Section from '../Section';
 import EditText from '../EditText';
+import axios from 'axios';
 import './LoginForm.css';
 
+const LOGIN_URL = 'http://localhost:3001/users/login';
+
 class LoginContent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+    };
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.onLogin = this.onLogin.bind(this);
+  }
+
+  onLogin() {
+    if (this.isInputValid()) {
+      //Send Request
+      axios
+        .post(LOGIN_URL, {
+          username: this.state.username,
+          password: this.state.password,
+        })
+        .then(response => {
+          // Got good response
+          console.log(response);
+          // TODO: change page, save student ID and name when backend is done
+          // temp for testing
+          this.setError('login successful');
+        })
+        .catch(loginError => {
+          // Invalid login
+          console.log(loginError);
+          var error = loginError.response.data.ErrorMessage;
+          if (
+            error === 'Password length must be between 8 and 64' ||
+            error === 'User does not exist' ||
+            error === 'Username must be alphanumeric'
+          ) {
+            this.setInvalidCredentials();
+          } else {
+            this.setError(error);
+          }
+        });
+    }
+  }
+
+  isInputValid() {
+    var id = this.state.username;
+    var password = this.state.password;
+    // Check if username & password are empty
+    if (this.isEmpty(id)) {
+      this.setError('Student ID is required.');
+      return false;
+    } else if (this.isEmpty(password)) {
+      this.setError('Password is required.');
+      return false;
+    } else if (!/^\d+$/.test(id)) {
+      // Invalid Student ID (contains letters)
+      this.setInvalidCredentials();
```

Comment:
Would it be better to display which field is causing the input to be invalid?

Reply:
"More secure" 😂